kola/tests/etcd/rolling.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package etcd
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/coreos/mantle/Godeps/_workspace/src/github.com/coreos/pkg/capnslog"
+	"github.com/coreos/mantle/kola/options"
+	"github.com/coreos/mantle/platform"
+)
+
+var (
+	// this block can be overridden with options flags
+	etcdVersion      = "etcd 2.0.12"
+	etcdVersion2     = `{"etcdserver":"2.1.0-alpha.1","etcdcluster":"2.1.0"}`
+	etcdBin          = "./etcd"
+	etcdBin2         = "./etcd2"
+	skipVersionCheck = false
+
+	dropPath    = "/home/core"
+	settingSize = 15 // number of random keys set and checked per node multiple times
+)
+
+func replaceDefaultsWithFlags(opts options.TestOptions) {
+	if opts.EtcdRollingVersion != "" {
+		etcdVersion = opts.EtcdRollingVersion
+	}
+	if opts.EtcdRollingVersion2 != "" {
+		etcdVersion2 = opts.EtcdRollingVersion2
+	}
+	if opts.EtcdRollingBin != "" {
+		etcdBin = opts.EtcdRollingBin
+	}
+	if opts.EtcdRollingBin2 != "" {
+		etcdBin2 = opts.EtcdRollingBin2
+	}
+	if opts.EtcdRollingSkipVersion {
+		skipVersionCheck = true
+	}
+
+}
+
+func RollingUpgrade(cluster platform.TestCluster) error {
+	replaceDefaultsWithFlags(options.Opts)
+
+	csize := len(cluster.Machines())
+
+	if plog.LevelAt(capnslog.DEBUG) {
+		// get journalctl -f from all machines before starting
+		for _, m := range cluster.Machines() {
+			if err := m.StartJournal(); err != nil {
+				return fmt.Errorf("failed to start journal: %v", err)
+			}
+		}
+	}
+
+	// drop in starting etcd binary
+	plog.Debug("adding files to cluster")
+	if err := cluster.DropFile(etcdBin); err != nil {
```

Comment:
perhaps add a TODO to gracefully skip tests if the binaries aren't available once there is a meaningful way to actually do that.